### PR TITLE
add build sfi index

### PR DESCRIPTION
**Pull request by cademirch**
Needed for https://github.com/brentp/mosdepth/issues/243

### CODE REVIEW

File: src/d4pkg/d4.nim [hunk 1]
```diff
@@ -241,3 +244,6 @@ when isMainModule:
   var o = d4f.values("chr2", 0, 60)
   echo o
   doAssert o[12] == 2'i32
+  doAssert d4.index_build_sfi("test.d4")
```
Comment (brentp) [inline src/d4pkg/d4.nim:247]: is there a way to test this has been created?

Comment (cademirch) [reply src/d4pkg/d4.nim:247]: Yeah I was thinking that and started working on it, but the  index type and check index function in the c api only supported the sum index, not sfi. Though now thinking about it again, i could just add a check sfi function. 

Other way to check which I've been using is to just call d4tools framedump and make sure the .index is there

Comment (brentp) [reply src/d4pkg/d4.nim:247]: ok. no need.

**Comment by brentp** [comment]
I tagged v0.0.5 with this change

**Pull request merged**